File: OnlineGroceryOrderingSystemAuthenticationJava/src/main/java/com/alpha/ogo/controller/LoginController.java
package com.alpha.ogo.controller;

import com.alpha.ogo.dto.CommonResponce;
import com.alpha.ogo.security.AuthenticationRequest;
import com.alpha.ogo.security.AuthenticationResponse;
import com.alpha.ogo.security.JwtUtil;
import com.alpha.ogo.security.MyUserDetails;
import com.alpha.ogo.service.impl.LoginServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;


@RestController
@CrossOrigin
public class LoginController {
	
	@Autowired
	private AuthenticationManager authenticationManager;

	@Autowired
	private JwtUtil jwtTokenUtil;

	@Autowired
	private LoginServiceImpl userDetailsService;
	
	@RequestMapping(value = "/authenticate", method = RequestMethod.POST)
	public CommonResponce<AuthenticationResponse> createAuthenticationToken(@RequestBody AuthenticationRequest authenticationRequest) throws Exception {
		CommonResponce<AuthenticationResponse> resopnce = new CommonResponce<AuthenticationResponse>();
		try {
			authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(authenticationRequest.getUserName(), authenticationRequest.getPassword()));
			
			final MyUserDetails userDetails = userDetailsService.loadUserByUsername(authenticationRequest.getUserName());
			final String jwt = jwtTokenUtil.generateToken(userDetails);
			
			resopnce.setCode(CommonResponce.successCode);
			resopnce.setMessage(CommonResponce.loginSuccessMessage);
			resopnce.setNewObject(new AuthenticationResponse(userDetails,jwt));
		}catch (BadCredentialsException e) {
			//throw new Exception("Incorrect username or password", e);
			resopnce.setCode(CommonResponce.errorCodeInvalidLogin);
			resopnce.setMessage(CommonResponce.errorMessageInvalidLogin);
		}
		
		return resopnce;
	}

}

File: OnlineGroceryOrderingSystemAuthentication/src/main/kotlin/com/allion/ogoss/controller/LoginController.kt
package com.allion.ogoss.controller

import com.allion.ogoss.domain.User
import com.allion.ogoss.dto.CommonResponce
import com.allion.ogoss.security.AuthenticationRequest
import com.allion.ogoss.security.AuthenticationResponse
import com.allion.ogoss.security.JwtUtil
import com.allion.ogoss.security.MyUserDetails
import com.allion.ogoss.service.impl.UserServiceImpl
import com.allion.ogoss.util.ControllerHelper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.BadCredentialsException
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class LoginController (private var userServiceImpl: UserServiceImpl,private var jwtTokenUtil: JwtUtil) :ControllerHelper(){

    @Autowired
    private val authenticationManager: AuthenticationManager? = null

    @GetMapping("hello")
    fun hello():String = "hello kotlin"

    @PostMapping("/authenticate")
    fun authentication(@RequestBody authenticationRequest : AuthenticationRequest) :CommonResponce<AuthenticationResponse>{
        val commonResponce= CommonResponce<AuthenticationResponse>()
        try {
        authenticationManager?.authenticate(UsernamePasswordAuthenticationToken(authenticationRequest.userName, authenticationRequest.password))
        val userDetails : MyUserDetails = userServiceImpl.loadUserByUsername(authenticationRequest.userName)
            println(userDetails.username)
            val jwt : String? = jwtTokenUtil.generateToken(userDetails);

            if(jwt !==null){
                val user:User = userServiceImpl.getUserDetails(userDetails.userName)

                commonResponce.code = successCode
                commonResponce.message = successMessage
                commonResponce.newObject = AuthenticationResponse(jwt,user,userDetails.authorities)
            }

        }catch (e : BadCredentialsException) {
            commonResponce.code = errorCodeInvalidLogin
            commonResponce.message = errorMessageInvalidLogin

        }catch ( ex: UsernameNotFoundException){
            println("value not found")
        }
        return commonResponce
    }

}